dotnet/src/Connectors/VectorData.Abstractions/RecordOptions/FilterOptions.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Microsoft.Extensions.VectorData;
+
+/// <summary>
+/// Defines options for filter search.
+/// </summary>
+/// <typeparam name="TRecord">Type of the record.</typeparam>
+public sealed class FilterOptions<TRecord>
+{
+    private int _skip = 0;
+
+    /// <summary>
+    /// Gets or sets the number of results to skip before returning results, that is, the index of the first result to return.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException">Thrown when the value is less than 0.</exception>
+    public int Skip
+    {
+        get => this._skip;
+        init
+        {
+            if (value < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), "Skip must be greater than or equal to 0.");
+            }
+
+            this._skip = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the data property to order by.
+    /// </summary>
+    /// <value>
+    /// If not provided the order of returned results can be non-deterministic.
+    /// </value>
+    public SortDefinition Sort { get; } = new();
```

Comment:
I'd align with LINQ (and SQL) naming here and call this OrderBy (and OrderByDefinition and OrderingDefinition), as in the comment summary.

(I think the reasoning was that sorting is slightly more associated with in-place sorting (i.e. doing something to the actual records inside the database), which is definitely not what this is about... But regardless of the reasons "OrderBy" is the thing everyone knows)

Reply:
This comment contradicts the other one: https://github.com/microsoft/semantic-kernel/pull/11112#discussion_r2029375995

I would keep the current naming as it's just plain simple and hard to misuse.

Reply:
Not quite - that comment was more about supporting Ascending/Descending rather than the OrderBy()/OrderByDescending() we originally discussed **inside** the lambda body. Let me make that clearer there and resolve here.